tests/rgen/Microsoft.Macios.Generator.Tests/Emitters/BindingSyntaxFactoryObjCRuntimeTests.cs

```diff
@@ -277,4 +277,76 @@ void GetStringAuxVariableTests (in Parameter parameter, string? expectedDeclarat
 			Assert.Equal (expectedDeclaration, declaration.ToString ());
 		}
 	}
+
+	class TestDataGetNSNumberAuxVariableTest : IEnumerable<object []> {
+		public IEnumerator<object []> GetEnumerator ()
+		{
+			yield return [
+				new Parameter (0, ReturnTypeForInt (), "myParam"),
+				"var nsb_myParam = NSNumber.FromInt32 (myParam);"
```

Comment:
you never ever feed the generated files in customers partial classes, right ? and you 100% control the typename and the namespace ? otherwise, it's safer to fullyqualify type names in sourcegen'd code

Reply:
You have a point. I'll do a second pass to do a fully qualified version of all the different types, I'll create a bug for it so that we use global:: etc